components/post_profile_picture/index.jsx

```diff
@@ -4,15 +4,18 @@
 import {connect} from 'react-redux';
 
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getUser, getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
 
 import PostProfilePicture from './post_profile_picture';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
     const config = getConfig(state);
 
     return {
         enablePostIconOverride: config.EnablePostIconOverride === 'true',
         hasImageProxy: config.HasImageProxy === 'true',
+        status: getStatusForUserId(state, ownProps.userId),
```

Comment:
See comments below, but would it make sense to leverage `initialOwnProps` instead?